functions/index.js

```diff
@@ -84,74 +69,67 @@ async function handleRequest(req, res, payloadHandler) {
 
   payload.token = token;
 
-  const ref = db.collection('rateLimits').doc(today).collection('tokens').doc(token);
-
-  let docExists = false;
-  let docData = {
-    attemptsCount: 0,
-    deliveredCount: 0,
-    errorCount: 0,
-    totalCount: 0,
-    expiresAt: getFirestoreTimestamp(),
-  };
-
+  // Create a rate limiter instance for this request
+  const rateLimiter = new RateLimiter(token, MAX_NOTIFICATIONS_PER_DAY, debug);
+  
+  let rateLimitInfo;
```

Comment:
You are only setting this once, this should be a const on L77. (GitHub does not allow me to suggest it)

Reply:
The try block means it'd be out of scope when its used later on in that case. It'd be nice if the try block was an expression that could return a value to avoid that.... 🤷🏻